src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +343,183 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
 			return ResponseEntity.status(HttpStatus.CREATED).body(dtos);
 		}
 	}
+	
+	/**
+	 * Retrieves all userSetting of the current user.
+	 * 
+	 * @return list of userSetting {@link UserSettingDTO}.
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/users/settings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<UserSettingDTO>> getUserSettings() throws OHServiceException {
+		LOGGER.info("Retrieve all userSettings of the current user");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		List<UserSetting> userSettings = userSettingManager.getUserSettingByUserName(userName);
+		if (userSettings == null || userSettings.isEmpty()) {
+		    LOGGER.info("No settings for the current user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		List<UserSettingDTO> userSettingsDTO = userSettingMapper.map2DTOList(userSettings);
+		LOGGER.info("Found {} user settings", userSettingsDTO);
+		return ResponseEntity.ok(userSettingsDTO);
+	}
+	
+	/**
+	 * Returns a {@link UserSettingDTO} of userSetting created.
+	 * 
+	 * @param userSettingDTO -  the {@link UserSettingDTO} to insert.
+	 * @return {@link UserSettingDTO} if the userSetting has been created, null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@PostMapping(value = "/users/settings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> newUserSettings(@Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Create a UserSetting");
+		String userName = userSettingDTO.getUser();
+		userSettingDTO.setId(0);
+		final String ADMIN = "admin";
+		if (!userName.equals(SecurityContextHolder.getContext().getAuthentication().getName()) 
```

Comment:
please extract `SecurityContextHolder.getContext().getAuthentication().getName())` into a local variable instead of executing twice in a rwo.

Reply:
okay already fix it on all methods that use that instruction.